packages/app/src/cli/services/context/id-matching.ts

```diff
@@ -51,6 +52,58 @@ function matchByNameAndType(
   return {matched: validMatches, pending: {local: pendingLocal, remote: pendingRemote}}
 }
 
+/**
+ * Automatically match local and remote sources if they have the same type and UID
+ */
+function matchByUID(
```

Comment:
In theory we shouldn't need to match anything right? just deploy with whatever UID is defined in the file. 
Or is this just to build the `diff` view?

Reply:
I think you are right, it doesn't make sense to ask for manual matching when the UID is different. But we still need to show the created/deleted ones in the diff. Let me refactor this. Cc: @amcaplan